### ISSUE TITLE
Startup and Deletion Issues with Slow Drives (network specifically)

### ISSUE THREAD

**Issue by da3dsoul** [d2phap/ImageGlass#462]
I'm a terrible person who Ctrl+A+Backspaces templates.

Firstly, I have not yet tried this on Moon, and after considering that, my version is a bit out of date, as well. I will update to Moon and report back after trying it for a while. I want this documented so I don't forget, though. These are multiple issues, but they are the same root cause and implementation, so I'm merging them.

When ImageGlass starts, it tries to load the image in the foreground, rather than the background. If the image is on a slow drive (network, flash drive, etc), then ImageGlass may fail to load it, then exit before giving it adequate time.

When going through a large list of images, deleting one, and auto-progressing to the next, it will sometimes not have the image loaded yet. It then just gives up and displays the load failed message, rather than waiting for the image to load.

When an image is loading, it freezes up and still takes input. When it catches up, it does everything at once.

All of these could be solved with asynchronous loading and a "loading wheel".

If this is all fixed, then cool.

Version 5.1.5.20 Installed locally
Windows 10 update something past 1803
Plenty of RAM and CPU
Images on Network Drive over Wi-Fi (because that's a great idea [/sarcasm]), mapped as Z:\

**Comment by da3dsoul** [comment]
After playing with the latest Moon for a while, I can verify that the issues are likely still present. I'll need to test more with startup and deleting.

**Comment by da3dsoul** [comment]
Okay, the issues with startup and auto-progression after deletion are "fixed". It still locks up the entire UI for several seconds while it loads the image, then does everything at once.

**Comment by d2phap** [comment]
thanks @da3dsoul  for the updates

**Comment by d2phap** [comment]
The new build v5.5.12.12 with asynchronous loading improvement:
https://imageglass.org/moon

There is known issue: the viewing image disappears while zooming the being loaded image.

**Comment by da3dsoul** [comment]
Yeah, it can use some work on notifying the user of "loading" as you mentioned. It takes a short while to update the title bar, probably while it figures out what the next picture is going to be (large folder). I sometimes speed through the folder, so the preloading doesn't have time to work. It could use a loading message there, as well, if it's doing that.
I got a "parameter is invalid" message when loading a picture. It was a normal picture, and backing out and then in again worked. I'll see if I can reproduce it. It was probably a rare race condition.
Other than that, it works well so far.

**Comment by da3dsoul** [comment]
Hmmm, I had an image give up on loading after updating the title bar with even the zoom scale (fit to window). It again worked when backing out and back in. I can verbose log and send it to you if you can tell me how.

**Comment by AnotherDimension-Ex** [comment]
Indeed, when you are viewing images fast some images skip loading, the only workaround it hit back button and next button to make the image load again.
I tested more than 600 images and I can reproduce every time.

<br>

![04223](https://user-images.githubusercontent.com/34074375/49908158-4b799000-fe8a-11e8-9e7a-954964c04da2.png)

**Comment by d2phap** [comment]
A new build to fix the below issues of Sync Loading: v5.5.12.15.
I need to more tests
https://imageglass.org/moon


> when you are viewing images fast some images skip loading, the only workaround it hit back button and next button to make the image load again.

> "parameter is invalid" message when loading a picture

**Comment by AnotherDimension-Ex** [comment]
> "parameter is invalid" message when loading a picture

Sorry, but I don't think I'm wrong this bug at less for me happens when viewing images very fast (keep pressing on next button ASAP)


Confirmed, this bug still exists in v5.5.12.15
I tested more than 400 images.


![04232](https://user-images.githubusercontent.com/34074375/50042807-d5d41680-0079-11e9-81df-ad19cea1395a.png)

![04234](https://user-images.githubusercontent.com/34074375/50042808-d5d41680-0079-11e9-8b3a-34f95984d73d.png)

**Comment by d2phap** [comment]
Hi @AnotherDimension-Ex 
In my test, it blank while loading next images, and after few seconds (depends on file size) the image came out. 

The known bug is, sometimes, when I zoom in/out, it disappears.

**Comment by d2phap** [comment]
Hi @AnotherDimension-Ex 
With this build, I think the issue is fixed. Please try v5.5.12.16
https://imageglass.org/moon
